Check tenant when reusing public net

jiriproX opened the pull request:
Signed-off-by: Jiri Prokes <jirix.x.prokes@intel.com>

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/taf3/taf/17)
<!-- Reviewable:end -->

jiriproX committed:
Check tenant when reusing public net

Signed-off-by: Jiri Prokes <jirix.x.prokes@intel.com>

jiriproX commented:
@rbbratta @mbanszex Please review it.

lijenni commented on taf/testlib/virtual_env.py:
```diff
@@ -28,7 +28,9 @@
-#  "mgmt_ip_cidr": "11.212.23.11/8",
+#  "external_net_ip_cidr": "11.212.23.0/24",
+#  "external_net_pool_start": "100",
+#  "external_net_pool_end": "199",
```
```
+#  "external_net_ip_cidr": "11.212.23.0/24",
 +#  "external_net_pool_start": "100",
 +#  "external_net_pool_end": "199",
```
Since generating the env json is done by our CI for taf testing, I think this review also needs a change in our CI, unless you can make this backwards compatible.

seungweonpark replied:
I think we'd better remove the ip info from source code, and move it to setup/configuration file for your own purpose.

lijenni replied:
The IPs here are just examples. They don't actually correspond to anything, so I don't think a change is necessary here.

rbbratta replied:
11.0.0.0/8 is owned by the DoD.  Do not use it as an example.

11.0.0.0/8	DoD Intel Information Systems	IANA Date 1993-05

jiriproX replied:
IP was already in code before my change. Anyway is it IP from private range  e.g.: 192.168.31.0/24 acceptable. It is in comment "required settings example".

rbbratta replied:
192.168 is okay or other RFC 1918 addresses are okay as long they don't match actual internal subnets.

mskrocki commented on taf/testlib/virtual_env.py:
```diff
@@ -248,7 +250,7 @@ def wait_for_server_status(self, vm_id, status):
         return waiters.wait_for_server_status(self.handle.servers_client, vm_id, status)
 
-    def ensure_public_access(self, try_reuse=False, networks_client=None, routers_client=None,
+    def ensure_public_access(self, try_reuse=True, networks_client=None, routers_client=None,
```
Defaults must stay as False

esmacgil replied:
If we need an alternate set of defaults, then define a different method that has the defaults that are needed and have it call this method.

lijenni replied:
try_reuse is defined by the pytest command line option --resuse-venv, so is it necessary to define an alternate set of defaults?

jiriproX replied:
Real default value here was never "False". Function is called with value taken from command line option --reuse-venv, which has default value "True". I just made it consistent.

mskrocki replied:
OK, keep True as default.

jiriproX replied:
Done

mskrocki commented on taf/testlib/virtual_env.py:
```diff
@@ -271,12 +273,17 @@ def ensure_public_access(self, try_reuse=False, networks_client=None, routers_cl
 
         # Try to reuse existing stuff that meets requirements, if desirable (devstack)
-        if try_reuse and self._reuse_public_access(net_ip, routers_client=routers_client):
+        if try_reuse and self._reuse_public_access(net_ip, routers_client=routers_client, tenant_id=tenant_id):
```
Why do we need tanant_id?

mskrocki commented on taf/testlib/virtual_env.py:
```diff
@@ -292,21 +299,24 @@ def ensure_public_access(self, try_reuse=False, networks_client=None, routers_cl
-            'enable_dhcp': False
-        }
-        self._create_subnet(public_network['id'], **subnet_kwargs)
+        if not public_network['subnets']:
```
we just created the network, how can it have a subnet?

mskrocki commented on taf/testlib/virtual_env.py:
```diff
@@ -28,7 +28,9 @@
 #  "id": "993",
 #  "images_share_path": "/mnt/berta/oses/openstack",
-#  "mgmt_ip_cidr": "11.212.23.11/8",
+#  "external_net_ip_cidr": "11.212.23.0/24",
```
leave the format as it was: 11.212.23.11/8, full cidr format shouldn't be required.

jiriproX replied:
external_net_ip_cidr was renamed to external_net_gw_ip_cidr as it have to contain gateway IP in cidr notation.
New example value:  # "external_net_gw_ip_cidr": "192.168.31.1/24"

esmacgil commented on taf/testlib/virtual_env.py:
```diff
@@ -107,7 +109,7 @@ def __init__(self, opts=None):
         self.settings = self._get_settings(self.opts.env)
         self.tempest_path = self.opts.tempest_path
-        self.reuse_venv = self.opts.reuse_venv
+        self.reuse_venv = eval(self.opts.reuse_venv)
```
Please avoid using eval.  If the value is JSON format, then use `json.loads()`, otherwise use `ast.literal_eval()`.  https://docs.python.org/3/library/ast.html#ast.literal_eval

jiriproX replied:
Done

esmacgil commented on taf/testlib/virtual_env.py:
```diff
@@ -271,12 +273,17 @@ def ensure_public_access(self, try_reuse=False, networks_client=None, routers_cl
+        _external_net_pool_start = self.settings.get('external_net_pool_start')
+        assert _external_net_pool_start
+        _external_net_pool_end = self.settings.get('external_net_pool_end')
+        assert _external_net_pool_end
```
These start and end values should have more testing/assertions to them, as in:

```
NET_POOL_MINIMUM = 100
NET_POOL_MAXIMUM = 254
...
        assert NET_POOL_MINIMUM <= _external_net_pool_start < _external_net_pool_end <= NET_POOL_MAXIMUM
```

jiriproX replied:
I did not find those two constants.  Should I put them to virtual_env code? Why is minimum = 100 ?

esmacgil replied:
Yes, please add any constants instead of using magic numbers.  The value 100 seemed to be the previous minimum, see line 304 of the previous revision and line 305 for the value 254.  If we have a reason to open up this range, then make those values instead of mine.

esmacgil commented on taf/testlib/virtual_env.py:
```diff
@@ -325,25 +335,29 @@ def ensure_public_access(self, try_reuse=False, networks_client=None, routers_cl
+                if ext_gw_info:
+                    net_id = ext_gw_info.get('network_id')
+                    if net_id and net_id in nets:
+                        net_2_router_map[net_id] = router['id']
```
Why not use an empty dictionary as the default value to the middle look up?

```
if tenant_id:
    for router in routers_resp['routers']:
        if tenant_id != router.get('tenant_id'):
            continue
        net_id = router.get('external_gateway_info', {}).get('network_id')
        net = nets.get(net_id)
        if net_id and net:
            net_2_router_map[net_id] = router['id']
```

jiriproX replied:
Code was rewritten.

esmacgil replied:
I missed the fact that `net` in my suggestion is only used for the conditional expression, so we don't need the `net = nets.get('net_id')` and the conditional expression can be as it has been: `if net_id and net_id in nets`

esmacgil commented on taf/testlib/virtual_env.py:
```diff
@@ -353,33 +367,41 @@ def _delete_external_elements(self, routers_client=None, networks_client=None,
-                        _net_cfg.public_router_id, _net_cfg.public_network_id = router_id, net_id
-                        return True
+        net_2_router_map = self._get_external_elements(routers_client=routers_client, tenant_id=tenant_id)
+        if net_2_router_map:
```
If `net_2_router_map` is an empty dictionary, then the for loop within this conditional's true block will immediately exit.  Please don't test for empty dictionaries immediately before looping over them.

If `net_2_router_map` can be None, then we need to change that.

jiriproX replied:
Done

esmacgil commented on taf/testlib/virtual_env.py:
```diff
@@ -353,33 +367,41 @@ def _delete_external_elements(self, routers_client=None, networks_client=None,
+                    for ip in ext_ips:
+                        if ip['ip_address'] in external_net:
+                            _net_cfg.public_router_id, _net_cfg.public_network_id = router_id, net_id
+                            return True
```
Please `continue` when possible, instead of nesting conditionals.

Please avoid repeating exactly similar function calls on successive lines.

```
for net_id, router_id in net_2_router_map.items():
    if not router_id or not tenant_id:
        continue
    rt_obj = routers_client.show_router(router_id)['router']
    if tenant_id != rt_obj['tenant_id']:
        continue
    ext_ips = rt_obj['external_gateway_info']['external_fixed_ips']
    with suppress(StopIteration):
        _net_cfg.public_router_id, _net_cfg.public_network_id = next((router_id, net_id) for ip in ext_ips if ip['ip_address'] in external_net)
        return True
```

jiriproX replied:
Done

esmacgil commented on taf/testlib/virtual_env.py:
```diff
@@ -577,9 +599,12 @@ def create_public_network(self, routers_client=None, networks_client=None, ports
+                'tenant_id': tenant_id,
             }
-            self._delete_external_elements(**del_external_kwargs)
+            external_net_id=self._delete_external_elements(**del_external_kwargs)
```
The assignment operator should be padded with white space:

```x = y```

jiriproX replied:
Done.

seungweonpark commented:
Well, still we don’t want to have IP. Just use xxx.xxx.xxx.xxx.

From: Jennifer Li [mailto:notifications@github.com]
Sent: Thursday, January 26, 2017 4:15 PM
To: taf3/taf <taf@noreply.github.com>
Cc: Park, Seungweon <seungweon.park@intel.com>; Comment <comment@noreply.github.com>
Subject: Re: [taf3/taf] Check tenant when reusing public net (#17)


@lijenni commented on this pull request.

________________________________

In taf/testlib/virtual_env.py<https://github.com/taf3/taf/pull/17>:

> @@ -28,7 +28,9 @@

 #  "instance_type": "settings",

 #  "id": "993",

 #  "images_share_path": "/mnt/berta/oses/openstack",

-#  "mgmt_ip_cidr": "11.212.23.11/8",

+#  "external_net_ip_cidr": "11.212.23.0/24",

+#  "external_net_pool_start": "100",

+#  "external_net_pool_end": "199",

The ip information are just examples. They don't actually correspond to anything.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://github.com/taf3/taf/pull/17>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AV944N07cjW8iv7iHQOujlHAp1Ahh5rkks5rWTb6gaJpZM4LsEMq>.

mskrocki commented on taf/testlib/virtual_env.py:
```diff
@@ -292,21 +299,24 @@ def ensure_public_access(self, try_reuse=False, networks_client=None, routers_cl
+            subnet_name = self.tempest_lib.common.utils.data_utils.rand_name('tempest-public-subnet')
+            allocation_prefix = _external_net_ip_cidr.rsplit('.', 1)[0]
+            net_ip_gw = net_ip
+            net_ip_gw.value = net_ip.first + 1
```
This will not scale, please revert to net_ip.ip.

jiriproX replied:
Done.

mskrocki commented on taf/testlib/virtual_env.py:
```diff
@@ -353,33 +367,41 @@ def _delete_external_elements(self, routers_client=None, networks_client=None,
-                    if ip['ip_address'] in mgmt_net:
-                        _net_cfg.public_router_id, _net_cfg.public_network_id = router_id, net_id
-                        return True
+        net_2_router_map = self._get_external_elements(routers_client=routers_client, tenant_id=tenant_id)
```
we have admin rights, we want to get all the public networks, not only from a specific tenant_id.

jiriproX replied:
Yes, we are getting all public networks. Nevertheless we need to check if router for corresponding tenant exist. Public network is visible across all tenants. Routers are tenant specific.

esmacgil commented:
Thank you for the comments, but where is the latest code?

jiriproX committed:
Check tenant when reusing public net
- update with review comments implementation

Signed-off-by: Jiri Prokes <jirix.x.prokes@intel.com>

jiriproX committed:
Check tenant when reusing public net
 - update with review comments implementation
 - new _reuse_public_network function was added

Signed-off-by: Jiri Prokes <jirix.x.prokes@intel.com>

esmacgil commented on taf/testlib/virtual_env.py:
```diff
@@ -420,6 +442,30 @@ def _reuse_public_access(self, external_net, routers_client=None, tenant_id=None
+        for network_id, router_id in net_2_router_map.items():
+            net_network = networks_client.show_network(network_id)['network']
+            for sub_net in net_network['subnets']:
+                if external_net and self.handle.os_adm.subnets_client.show_subnet(sub_net)['subnet']['gateway_ip'] in external_net:
```
Why not check `external_net` before doing all this looping?  AFAICT, not having an `external_net` is as bad as not having a `tenant_id`.

Then the rest looks like a `next()` statement to me:

```
def network_test(net_id):
  subnet = self.handle.os_adm.subnets_client.show_subnet(net_id['subnet'])
  return external_net_set.intersection(set(subnet['subnet']['gateway_ip']))

def log_and_return(template, value, *args, **kwargs):
  if value:
    self.class_logger.debug(template, value, *args, **kwargs)
  return value

external_net_set = set(external_net)
net_2_router_map = self._get_external_elements(...)
iter1 = (network_id for network_id in net_2_router_map if network_test(network_id))
return log_and_return('...', next(iter1, None))
```

jiriproX replied:
I've moved check for external_net to the beginning. I've removed router_id which were not used.
Rest remain the same as proposed solution is not working as it is. 
show_subnet needs subnet_id,  net_id does not contain 'subnet' as it is just Id. I replaced net_id with corresponding network.
Intersection provides empty set. There seems to be wrong type of object to intersect with.
I stop next further investigation here. Original solution was tested and it is working.

esmacgil commented on taf/testlib/virtual_env.py:
```diff
@@ -291,22 +292,37 @@ def ensure_public_access(self, try_reuse=True, networks_client=None, routers_cli
+                'tenant_id': tenant_id,
+            }
+            public_network = self.create_public_network(**public_network_kwargs)
+            assert public_network
```
please avoid asserting something that will be more tested by an upcoming statement.  i hope we are not using AssertionErrors as exceptions to recover from.

jiriproX replied:
Done

jiriproX committed:
Check tenant when reusing public net
- implementation of review comments

Signed-off-by: Jiri Prokes <jirix.x.prokes@intel.com>

esmacgil commented:
Reviewed 1 of 2 files at r2, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 12 unresolved discussions, some commit checks failed.

---

*[taf/testlib/virtual_env.py, line 302 at r1](https://reviewable.io:443/reviews/taf3/taf/17#-Kc_DPidrYf-7FDGfS9R-r1-302:-KcdJ5-ysOb35eX00-_t:b-wdy9mi) ([raw file](https://github.com/taf3/taf/blob/73078350f384afaf0358bb476efca45e58b82dd8/taf/testlib/virtual_env.py#L302)):*
<details><summary><i>Previously, mskrocki (Maciej Skrocki) wrote…</i></summary><blockquote>

we just created the network, how can it have a subnet?
</blockquote></details>

We may have just reused an existing network (see line 300), notice how the logging about "No reuse" is now inside an `else` block.

---

*[taf/testlib/virtual_env.py, line 461 at r2](https://reviewable.io:443/reviews/taf3/taf/17#-Kc_DPidrYf-7FDGfS9R-r2-461:-KcdH87VQoKoR0Y_TlJh:b-r786dg) ([raw file](https://github.com/taf3/taf/blob/a62ca1b8af2bc9d532997c2ca101977104cabd9b/taf/testlib/virtual_env.py#L461)):*
<details><summary><i>Previously, jiriproX wrote…</i></summary><blockquote>

I've moved check for external_net to the beginning. I've removed router_id which were not used.
Rest remain the same as proposed solution is not working as it is. 
show_subnet needs subnet_id,  net_id does not contain 'subnet' as it is just Id. I replaced net_id with corresponding network.
Intersection provides empty set. There seems to be wrong type of object to intersect with.
I stop next further investigation here. Original solution was tested and it is working. 
</blockquote></details>

OK, yes, my suggestion was not fully baked and it was further than needed.  I do think that the `log_and_return` method is something that is generally useful.  In some cases this is done with a decorator, but that may not work/be appropriate in all cases.  Thank you for doing the other suggestions.

---

*[taf/testlib/virtual_env.py, line 373 at r3](https://reviewable.io:443/reviews/taf3/taf/17#-KcdF-o3mtJ09zV6Nt96:-KcdF-o3mtJ09zV6Nt97:b-vdoozp) ([raw file](https://github.com/taf3/taf/blob/efa094a5a49a007651a559365d5d2542a7fe354f/taf/testlib/virtual_env.py#L373)):*
> ```Python
>                 continue
>             net_id = ext_gw_info.get('network_id')
>             if net_id and net_id in nets:
> ```

This looks like another continue and it can be combined with the previous lines:

```
net_id = router.get('external_gateway_info', {}).get('network_id')
if not net_id or net_id not in nets:
    continue
if tenant_id and tenant_id == router.get('tenant_id'):
    self.net_2_router_map[net_id] = router['id']
elif net_id not in self.net_2_router_map:
    self.net_2_router_map[net_id] = None
```

---

*[taf/testlib/virtual_env.py, line 377 at r3](https://reviewable.io:443/reviews/taf3/taf/17#-KcdEmCJBr4QPKgEcHvh:-KcdEmCJBr4QPKgEcHvi:blx3fp6) ([raw file](https://github.com/taf3/taf/blob/efa094a5a49a007651a559365d5d2542a7fe354f/taf/testlib/virtual_env.py#L377)):*
> ```Python
>                     self.net_2_router_map[net_id] = router['id']
>                 else:
>                     if not net_id in self.net_2_router_map:
> ```

Why not combine this `if` with the previous `else` for an `elif`?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/taf3/taf/17)*
<!-- Sent from Reviewable.io -->

jiriproX commented:
Review status: all files reviewed at latest revision, 11 unresolved discussions, some commit checks failed.

---

*[taf/testlib/virtual_env.py, line 31 at r1](https://reviewable.io:443/reviews/taf3/taf/17#-Kc_DPidrYf-7FDGfS9R-r1-31:-KcrSmO4mBHoc3M7xZhx:b-896fix) ([raw file](https://github.com/taf3/taf/blob/73078350f384afaf0358bb476efca45e58b82dd8/taf/testlib/virtual_env.py#L31)):*
<details><summary><i>Previously, jiriproX wrote…</i></summary><blockquote>

external_net_ip_cidr was renamed to external_net_gw_ip_cidr as it have to contain gateway IP in cidr notation.
New example value:  # "external_net_gw_ip_cidr": "192.168.31.1/24"
</blockquote></details>

Done.

---

*[taf/testlib/virtual_env.py, line 33 at r1](https://reviewable.io:443/reviews/taf3/taf/17#-Kc_DPidrYf-7FDGfS9R-r1-33:-KcrT3evIpCmWWZKS0Ho:b-896fix) ([raw file](https://github.com/taf3/taf/blob/73078350f384afaf0358bb476efca45e58b82dd8/taf/testlib/virtual_env.py#L33)):*
<details><summary><i>Previously, rbbratta (Ross Brattain) wrote…</i></summary><blockquote>

192.168 is okay or other RFC 1918 addresses are okay as long they don't match actual internal subnets.
</blockquote></details>

Done.

---

*[taf/testlib/virtual_env.py, line 253 at r1](https://reviewable.io:443/reviews/taf3/taf/17#-Kc_DPidrYf-7FDGfS9R-r1-253:-KcrT7hjLfbs4OHUO0j2:b-896fix) ([raw file](https://github.com/taf3/taf/blob/73078350f384afaf0358bb476efca45e58b82dd8/taf/testlib/virtual_env.py#L253)):*
<details><summary><i>Previously, jiriproX wrote…</i></summary><blockquote>

Done
</blockquote></details>

Done.

---

*[taf/testlib/virtual_env.py, line 283 at r1](https://reviewable.io:443/reviews/taf3/taf/17#-Kc_DPidrYf-7FDGfS9R-r1-283:-KcrTSD7OE9YuhaUL0H3:b-896fix) ([raw file](https://github.com/taf3/taf/blob/73078350f384afaf0358bb476efca45e58b82dd8/taf/testlib/virtual_env.py#L283)):*
<details><summary><i>Previously, esmacgil (Edward MacGillivray) wrote…</i></summary><blockquote>

Yes, please add any constants instead of using magic numbers.  The value 100 seemed to be the previous minimum, see line 304 of the previous revision and line 305 for the value 254.  If we have a reason to open up this range, then make those values instead of mine.
</blockquote></details>

Done.

---

*[taf/testlib/virtual_env.py, line 286 at r1](https://reviewable.io:443/reviews/taf3/taf/17#-Kc_DPidrYf-7FDGfS9R-r1-286:-KcrUVYEdpKCuSThLh0W:b-fg946p) ([raw file](https://github.com/taf3/taf/blob/73078350f384afaf0358bb476efca45e58b82dd8/taf/testlib/virtual_env.py#L286)):*
<details><summary><i>Previously, mskrocki (Maciej Skrocki) wrote…</i></summary><blockquote>

Why do we need tanant_id?
</blockquote></details>

We were getting all public networks and all routers. Nevertheless we need to check if router for corresponding tenant exist. Public network is visible across all tenants. Routers are tenant specific.

---

*[taf/testlib/virtual_env.py, line 302 at r1](https://reviewable.io:443/reviews/taf3/taf/17#-Kc_DPidrYf-7FDGfS9R-r1-302:-KcrVGtQaHR1DgjuBvTG:by6u5td) ([raw file](https://github.com/taf3/taf/blob/73078350f384afaf0358bb476efca45e58b82dd8/taf/testlib/virtual_env.py#L302)):*
<details><summary><i>Previously, esmacgil (Edward MacGillivray) wrote…</i></summary><blockquote>

We may have just reused an existing network (see line 300), notice how the logging about "No reuse" is now inside an `else` block.
</blockquote></details>

Yes, in case of reused public network subnet is  already present.

---

*[taf/testlib/virtual_env.py, line 306 at r1](https://reviewable.io:443/reviews/taf3/taf/17#-Kc_DPidrYf-7FDGfS9R-r1-306:-KcrTZsaAKMfQTGAOdad:b-896fix) ([raw file](https://github.com/taf3/taf/blob/73078350f384afaf0358bb476efca45e58b82dd8/taf/testlib/virtual_env.py#L306)):*
<details><summary><i>Previously, jiriproX wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Done.

---

*[taf/testlib/virtual_env.py, line 352 at r1](https://reviewable.io:443/reviews/taf3/taf/17#-Kc_DPidrYf-7FDGfS9R-r1-352:-KcrWAUiy990kWUjGBvJ:b-896fix) ([raw file](https://github.com/taf3/taf/blob/73078350f384afaf0358bb476efca45e58b82dd8/taf/testlib/virtual_env.py#L352)):*
<details><summary><i>Previously, esmacgil (Edward MacGillivray) wrote…</i></summary><blockquote>

I missed the fact that `net` in my suggestion is only used for the conditional expression, so we don't need the `net = nets.get('net_id')` and the conditional expression can be as it has been: `if net_id and net_id in nets`
</blockquote></details>

Done.

---

*[taf/testlib/virtual_env.py, line 373 at r3](https://reviewable.io:443/reviews/taf3/taf/17#-KcdF-o3mtJ09zV6Nt96:-KcrTddE9uo5v_QGNGBI:b-896fix) ([raw file](https://github.com/taf3/taf/blob/efa094a5a49a007651a559365d5d2542a7fe354f/taf/testlib/virtual_env.py#L373)):*
<details><summary><i>Previously, esmacgil (Edward MacGillivray) wrote…</i></summary><blockquote>

This looks like another continue and it can be combined with the previous lines:

```
net_id = router.get('external_gateway_info', {}).get('network_id')
if not net_id or net_id not in nets:
    continue
if tenant_id and tenant_id == router.get('tenant_id'):
    self.net_2_router_map[net_id] = router['id']
elif net_id not in self.net_2_router_map:
    self.net_2_router_map[net_id] = None
```
</blockquote></details>

Done.

---

*[taf/testlib/virtual_env.py, line 377 at r3](https://reviewable.io:443/reviews/taf3/taf/17#-KcdEmCJBr4QPKgEcHvh:-KcrTwly2zOA8of3Mtp8:b-896fix) ([raw file](https://github.com/taf3/taf/blob/efa094a5a49a007651a559365d5d2542a7fe354f/taf/testlib/virtual_env.py#L377)):*
<details><summary><i>Previously, esmacgil (Edward MacGillivray) wrote…</i></summary><blockquote>

Why not combine this `if` with the previous `else` for an `elif`?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/taf3/taf/17)*
<!-- Sent from Reviewable.io -->

jiriproX committed:
Check tenant when reusing public net
  - implementation of review comments

Signed-off-by: Jiri Prokes <jirix.x.prokes@intel.com>

jiriproX committed:
Check tenant when reusing public net
  - correction of checker errors

Signed-off-by: Jiri Prokes <jirix.x.prokes@intel.com>

esmacgil commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/taf3/taf/17#-:-Kcs_B6zbp64vTkBaCAT:bnfp4nl)*
<!-- Sent from Reviewable.io -->

rbbratta commented:
Reviewed 1 of 2 files at r2, 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/taf3/taf/17)*
<!-- Sent from Reviewable.io -->

seungweonpark commented:
Reviewed 1 of 2 files at r2, 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/taf3/taf/17)*
<!-- Sent from Reviewable.io -->

seungweonpark commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: all files reviewed at latest revision, 5 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/taf3/taf/17#-:-KctyZQYNXO0x_qClYNn:bnfp4nl)*
<!-- Sent from Reviewable.io -->

mskrocki commented:
-2

---

Review status: all files reviewed at latest revision, 5 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/taf3/taf/17#-:-Kcu1JV5UYXDK0moVEm6:b32v6y)*
<!-- Sent from Reviewable.io -->

mskrocki commented:
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[taf/testlib/virtual_env.py, line 328 at r5](https://reviewable.io:443/reviews/taf3/taf/17#-KcuD38TYxwedRpYiBbv:-KcuD38TYxwedRpYiBbw:b-i97fr9) ([raw file](https://github.com/taf3/taf/blob/1839aaee747a0ba50cfc3ba76f95f615980f391e/taf/testlib/virtual_env.py#L328)):*
> ```Python
>         return waiters.wait_for_server_status(self.handle.servers_client, vm_id, status)
> 
>     #TODO: JP check if create_external_router is needed
> ```

For this review to move forward we need to make this investigation: Do we need to create router at all? if not, then we just need to simplify.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/taf3/taf/17)*
<!-- Sent from Reviewable.io -->

rbbratta commented:
Review scrub: need confirmation from Martin B

---

Review status: all files reviewed at latest revision, 5 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/taf3/taf/17#-:-Kd7plndEOg9L0Ml4qrW:b9lrrkw)*
<!-- Sent from Reviewable.io -->

jiriproX committed:
Check tenant when reusing public net
  - removing create_external_router parameter, which came from rebase and is not needed any longer

Signed-off-by: Jiri Prokes <jirix.x.prokes@intel.com>

esmacgil commented:
-1

---

Reviewed 1 of 1 files at r6.
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[taf/testlib/virtual_env.py, line 419 at r6](https://reviewable.io:443/reviews/taf3/taf/17#-KdCA6xb5jWtU_MqElLp:-KdCA6xb5jWtU_MqElLq:b5y3swd) ([raw file](https://github.com/taf3/taf/blob/0112087e19952efad0f1396a603f9faf43e1d1c3/taf/testlib/virtual_env.py#L419)):*
> ```Python
>         _net_cfg.public_router_id = public_router['id']
> 
>         if _net_cfg.public_network_id and _net_cfg.public_router_id:
> ```

Please return the result of this evaluation instead of the literal values:

```
return _net_cfg.public_network_id and _net_cfg.public_router_id
```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/taf3/taf/17#-:-KdCArjy_E0_JPLiTkNf:b32v6x)*
<!-- Sent from Reviewable.io -->

lijenni commented:
review scrub: what's the status on this?

esmacgil commented:
I requested a change:
* [virtual_env.py:419](https://github.com/taf3/taf/blob/0112087e19952efad0f1396a603f9faf43e1d1c3/taf/testlib/virtual_env.py#L419)

And there are several discussions that are pending Maciej's assent to:
* [virtual_env.py:31](https://github.com/taf3/taf/blob/0112087e19952efad0f1396a603f9faf43e1d1c3/taf/testlib/virtual_env.py#L31)
* [virtual_env.py:359](https://github.com/taf3/taf/blob/0112087e19952efad0f1396a603f9faf43e1d1c3/taf/testlib/virtual_env.py#L359)
* [virtual_env.py:389](https://github.com/taf3/taf/blob/0112087e19952efad0f1396a603f9faf43e1d1c3/taf/testlib/virtual_env.py#L389)
* [virtual_env.py:492](https://github.com/taf3/taf/blob/0112087e19952efad0f1396a603f9faf43e1d1c3/taf/testlib/virtual_env.py#L492)

so one of the following needs to happen:
* Maciej assents
* we need to provide more evidence that he should assent
* the code needs to be changed so that he can assent

The reason I know this is because (on Reviewable) each discussion thread has
* a set of circles in the lower-right corner of the discussion box
* the top circle shows the active user's level of engagement, one of:
  * ellipsis: following but neutral on resolving
  * check mark in dark circle: satisfied ("OK"), what i'm calling assent
  * check mark in light circle: discussing and not blocking
  * hyphen in dark circle: blocking
  * red circle: discussing but not satisfied
    * this is not a state in the list, but is the result of starting a discussion
* click on the top circle and the other circles will
  * expand leftward showing the state of other users
  * allow the current user to set their state

The current user can also alter their state by commenting in the discussion.

---

Review status: all files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[taf/testlib/virtual_env.py, line 419 at r6](https://reviewable.io:443/reviews/taf3/taf/17#-KdCA6xb5jWtU_MqElLp:-KfOe3FmOIyxhDS-N0XF:b-lnhgvp) ([raw file](https://github.com/taf3/taf/blob/0112087e19952efad0f1396a603f9faf43e1d1c3/taf/testlib/virtual_env.py#L419)):*
<details><summary><i>Previously, esmacgil (Edward MacGillivray) wrote…</i></summary><blockquote>

Please return the result of this evaluation instead of the literal values:

```
return _net_cfg.public_network_id and _net_cfg.public_router_id
```
</blockquote></details>

Still waiting for this change

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/taf3/taf/17#-:-KfOYnMvU4J2gt3rDkDr:b-p6k0ij)*
<!-- Sent from Reviewable.io -->

rbbratta commented:
Closing this, changes done in Maciej's pull request #33
